src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,24 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.writeLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        lock.readLock().lock();
+        try {
+            return list.get(index);
+        } finally {
+            lock.readLock().unlock();
+        }
     }
 
     public int size() {
-        // write your code here
-        return 0;
+        return list.size();
```

Comment:
The `size` method should also use a read lock to ensure thread safety when accessing the list size. Consider wrapping the return statement with a read lock, similar to the `get` method.